### PR TITLE
[SERVICES-2155] context tracker

### PR DESCRIPTION
**Pull request by claudiulataretu**
## Reasoning
- use timestamp to query data from deep history
  
## Proposed Changes
- added context tracker from nestjs-sdk
- use context tracker to propagate timestamp across services
- query data with timestamp

## How to test
- any query with `timestamp` in request headers should return data from the past

### CODE REVIEW

File: src/utils/logging.interceptor.ts [hunk 1]
```diff
@@ -23,8 +24,13 @@ export class LoggingInterceptor implements NestInterceptor {
             const { req } = gqlContext.getContext();
 
             let origin = 'Unknown';
+            let timesamp: number = undefined;
             if (req !== undefined) {
                 origin = req?.headers?.['origin'] ?? 'Unknown';
+                timesamp = req?.headers?.['timestamp'];
```
Comment (bogdan-rosianu) [inline src/utils/logging.interceptor.ts:30]: typo: timestamp*

File: src/services/multiversx-communication/mx.api.service.ts [hunk 2]
```diff
@@ -82,6 +83,15 @@ export class MXApiService {
     ): Promise<T> {
         const profiler = new PerformanceProfiler(`${name} ${resourceUrl}`);
         try {
+            const context = ContextTracker.get();
+            if (
+                this.apiConfigService.isDeephistoryActive() &&
+                context &&
+                context.deepHistoryTimestamp
+            ) {
+                resourceUrl = `${resourceUrl}&timestamp=${context.deepHistoryTimestamp}`;
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply src/services/multiversx-communication/mx.api.service.ts:92]: right. maybe we can use this: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append

File: src/helpers/proxy.network.provider.profiler.ts [hunk 3]
```diff
@@ -22,4 +33,17 @@ export class ProxyNetworkProviderProfiler extends ProxyNetworkProvider {
...
     }
+
+    async doPostGeneric(resourceUrl: string, payload: any): Promise<any> {
+        const context = ContextTracker.get();
+        if (
+            this.apiConfigService.isDeephistoryActive() &&
+            context &&
+            context.deepHistoryTimestamp
+        ) {
+            resourceUrl = `${resourceUrl}?timestamp=${context.deepHistoryTimestamp}`;
```
[the remark this answers is not in the document]
Comment (claudiulataretu) [reply src/helpers/proxy.network.provider.profiler.ts:44]: I'm not using proxy, but the MVX API which knows how to handle the timestamp

File: src/services/multiversx-communication/mx.api.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (claudiulataretu) [reply src/services/multiversx-communication/mx.api.service.ts]: done

File: src/utils/logging.interceptor.ts (hunk 1 above)
Comment (claudiulataretu) [reply src/utils/logging.interceptor.ts]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by mad2sm0key — APPROVED** [review]

**Pull request merged**